From 9c3bf5a16a92eecde8c9abb68773681bf8160548 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 16 Nov 2016 15:39:31 +0800
Subject: [PATCH 1323/5242] MLK-13577-2 ARM: imx7: mu: set ABF0 of MU_CR to
 notify M4 MU finish initialization

commit  9ed7abb41cff2d27ca3a2a3e2ba802b3c4488c7b from
https://source.codeaurora.org/external/imx/linux-imx.git

set ABF0 of MU_CR to let M4 know MU is ready, thus MU can initialize rpmsg
later.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
(cherry picked from commit 4969903fc8d32c5e6dfc4fe5f2f68e52aef080bc)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/mu.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/mu.c b/arch/arm/mach-imx/mu.c
index fd4c58f..6a1bfaf 100644
--- a/arch/arm/mach-imx/mu.c
+++ b/arch/arm/mach-imx/mu.c
@@ -408,9 +408,10 @@ static int imx_mu_probe(struct platform_device *pdev)
 
 	INIT_DELAYED_WORK(&mu_work, mu_work_handler);
 	INIT_DELAYED_WORK(&rpmsg_work, rpmsg_work_handler);
+	/* bit0 of MX7ULP_MU_CR used to let m4 to know MU is ready now */
 	if (cpu_is_imx7ulp())
 		writel_relaxed(readl_relaxed(mu_base + MX7ULP_MU_CR) |
-			BIT(26) | BIT(27), mu_base + MX7ULP_MU_CR);
+			BIT(0) | BIT(26) | BIT(27), mu_base + MX7ULP_MU_CR);
 	else
 		writel_relaxed(readl_relaxed(mu_base + MU_ACR) |
 			BIT(26) | BIT(27), mu_base + MU_ACR);
-- 
1.7.9.5

